src/submodules/schedulers.jl

```diff
@@ -149,11 +158,11 @@ function by_property(p)
     end
 end
 
-struct ByProperty{P}
+mutable struct ByProperty{P}
     p::P
-    properties::Vector{Float64} # TODO: don't assume Float64
     ids::Vector{Int}
     perm::Vector{Int}
+    properties::Vector
```

Comment:
Doesn't this type instability bring a big performance hit...?

Reply:
this seems not to be the case, I calculated the performance with the scheduler benchmark and it seems fine, will post the result

Reply:
actually changed the implementation in its final form since saving properties n the struct is unnecessary, but anyway it didn't change the performance when I calculated it

Reply:
Let me know if you can think of a better method for handling `properties` (even if allocating doesn't seem to be problematic in the benchmarks)